public/assets/js/home.js

```diff
@@ -0,0 +1,91 @@
+
+function darkmode() {
+  var element = document.body;
+  var type = document.getElementById('mode');
+  element.classList.toggle("darkmode");
+  document.body.classList.toggle("dark-theme");
+  if (document.body.classList.contains("dark-theme")) {
+    type.innerHTML = "Light Mode";
+  }
+  else {
+    type.innerHTML = "Dark Mode";
+  }
+
+}
+
+  $('#scroll').click(function (e) {
+    e.preventDefault();
+    $('html, body').animate({ scrollTop: 0 }, '1000');
+  });
+
+$(window).scroll(function () {
+  var scroll = $(window).scrollTop();
+
+  if (scroll >= 300) {
+    $("#scroll").addClass("show");
+  } else {
+    $("#scroll").removeClass("show");
+  }
+});
+
+$(document).ready(() => {
+  $("#imagePreview").css("display", "none");
+  $("#create-post-media").change(function () {
+    const file = this.files[0];
+    if (file) {
+      let reader = new FileReader();
+      reader.onload = function (event) {
+        $("#imagePreview")
+          .attr("src", event.target.result);
+        $("#imagePreview").css("display", "block");
+      };
+      reader.readAsDataURL(file);
+    }
+  });
+});
```

Comment:
1/ What does this code do?
2/ 2here is the comment?
3/ Do you know how to use arrow functions or is this just copied form somewhere?

Reply:
1. This code is for previewing the image before it is uploaded.
2. I will add comments
3. Yes , i have worked with react previously so i am familiar with arrow functions. For the logic part i have taken help from stackoverflow. Source : https://stackoverflow.com/questions/4459379/preview-an-image-before-it-is-uploaded